File: count-partitions-with-even-sum-difference/cpwesd.rs
pub fn count_partitions(nums: Vec<i32>) -> i32 {
    let init = nums[0] - nums.get(1..).unwrap().into_iter().map(|n| *n).sum::<i32>();
    if init % 2 == 0 {
        return (nums.len() - 1) as i32;
    } else {
        return 0;
    }
}

fn main() {
    dbg!(count_partitions(vec![10, 10, 3, 7, 6]));
    dbg!(count_partitions(vec![1, 2, 2]));
    dbg!(count_partitions(vec![2, 4, 6, 8]));
    dbg!(count_partitions(vec![1, 5, 5]));
}

File: count-partitions-with-even-sum-difference/cpwesd.go
package main

import "fmt"

func sum(init int, n ...int) int {
	for _, nn := range n {
		init += nn
	}

	return init
}

func countPartitions(nums []int) int {
	x := sum(nums[0], nums[1:]...)
	if x%2 != 0 {
		return 0
	}

	return len(nums) - 1
}

func main() {
	fmt.Println(countPartitions([]int{10, 10, 3, 7, 6}))
	fmt.Println(countPartitions([]int{1, 2, 2}))
	fmt.Println(countPartitions([]int{2, 4, 6, 8}))
}
